clickhouse_options.go

```diff
@@ -166,12 +195,12 @@ func (o *Options) fromDSN(in string) error {
 	}
 	o.scheme = dsn.Scheme
 	switch dsn.Scheme {
-	case "http":
+	case HTTP.String():
```

Comment:
Probably we can leave "http" and "https" here as string literals, and keep only `HTTP` protocol.

Because https is "secure HTTP", same as TLS is "secure native", just for consistency.

So, IMO it sould be "HTTPS, HTTP, Native, SecureNative" or "HTTP, Native" and "secure / not secure"

Reply:
yes this has bugged me as well. For native we use the presence of TLS config to denote a secure connection i.e. https://github.com/ClickHouse/ClickHouse-go#tlsssl
Let's drop HTTPS and detect using SSL for consistency.